state/in_memory_store_test.go

```diff
@@ -106,7 +106,240 @@ func Test_InMemoryStore_GetConfig(t *testing.T) {
+			},
+		}
+		finalizedConf.Finalize()
+		store := NewInMemoryStore(finalizedConf)
+
+		actual, exists := store.GetTask("task_a")
+		assert.True(t, exists)
+
+		assert.Len(t, *store.conf.Tasks, 1)
+		storedConf := (*store.conf.Tasks)[0]
 
+		// Confirm returned config has same value as stored
+		assert.Equal(t, *storedConf, actual)
+
+		// Confirm returned config references different object from stored
+		assert.NotSame(t, storedConf, &actual)
+
+		// Confirm returned config field reference different object from stored
+		assert.NotSame(t, storedConf.Name, actual.Name)
+	})
+}
+
+func Test_InMemoryStore_SetTask(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name      string
+		stateConf *config.Config
+		input     config.TaskConfig
+		expected  config.TaskConfigs
+	}{
+		{
+			name: "update existing task",
+			stateConf: &config.Config{
+				Tasks: &config.TaskConfigs{
+					{Name: config.String("existing_task")},
+				},
+			},
+			input: config.TaskConfig{
+				Name:        config.String("existing_task"),
+				Description: config.String("new description"),
+			},
+			expected: config.TaskConfigs{
+				&config.TaskConfig{
+					Name:        config.String("existing_task"),
+					Description: config.String("new description"),
+				},
+			},
+		},
+		{
+			name: "add new task",
+			stateConf: &config.Config{
+				Tasks: &config.TaskConfigs{
+					{Name: config.String("existing_task")},
+				},
+			},
+			input: config.TaskConfig{
+				Name: config.String("new_task"),
+			},
+			expected: config.TaskConfigs{
+				&config.TaskConfig{
+					Name: config.String("existing_task"),
+				},
+				&config.TaskConfig{
+					Name: config.String("new_task"),
+				},
+			},
+		},
+		{
+			name:      "no prior existing tasks",
+			stateConf: &config.Config{},
+			input: config.TaskConfig{
+				Name: config.String("new_task"),
+			},
+			expected: config.TaskConfigs{
+				&config.TaskConfig{
+					Name: config.String("new_task"),
+				},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc.stateConf.Finalize()
+			store := NewInMemoryStore(tc.stateConf)
+
+			// finalize the task configs
+			bp := tc.stateConf.BufferPeriod
+			wd := config.StringVal(tc.stateConf.WorkingDir)
+			tc.input.Finalize(bp, wd)
+			tc.expected.Finalize(bp, wd)
+
+			store.SetTask(tc.input)
+			assert.Equal(t, tc.expected, *store.conf.Tasks)
+		})
+	}
+}
+
+func Test_InMemoryStore_DeleteTask(t *testing.T) {
+	t.Parallel()
+
+	cases := []struct {
+		name     string
+		input    string
+		expected config.TaskConfigs
+	}{
+		{
+			"task exists",
+			"existing_task",
+			config.TaskConfigs{},
+		},
+		{
+			"task does not exist",
+			"non_existing_task",
+			config.TaskConfigs{
+				{Name: config.String("existing_task")},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			conf := &config.Config{
+				Tasks: &config.TaskConfigs{
+					{Name: config.String("existing_task")},
```

Comment:
Could you add the case of removing a task when there are multiple existing tasks?

Reply:
Sure, added